edge-agent/src/Microsoft.Azure.Devices.Edge.Agent.Kubernetes/planners/KubernetesPlanner.cs

```diff
@@ -18,22 +18,25 @@ public class KubernetesPlanner : IPlanner
     {
         readonly IKubernetes client;
         readonly ICommandFactory commandFactory;
-        readonly ICombinedConfigProvider<CombinedDockerConfig> combinedConfigProvider;
         readonly string deviceNamespace;
         readonly ResourceName resourceName;
+        readonly ICombinedConfigProvider<CombinedDockerConfig> dockerConfigProvider;
+        readonly ICombinedConfigProvider<CombinedKubernetesConfig> kubernetesConfigProvider;
 
         public KubernetesPlanner(
             string deviceNamespace,
             ResourceName resourceName,
             IKubernetes client,
             ICommandFactory commandFactory,
-            ICombinedConfigProvider<CombinedDockerConfig> combinedConfigProvider)
+            ICombinedConfigProvider<CombinedDockerConfig> dockerConfigProvider,
+            ICombinedConfigProvider<CombinedKubernetesConfig> kubernetesConfigProvider)
         {
             this.resourceName = Preconditions.CheckNotNull(resourceName, nameof(resourceName));
             this.deviceNamespace = Preconditions.CheckNonWhiteSpace(deviceNamespace, nameof(deviceNamespace));
             this.client = Preconditions.CheckNotNull(client, nameof(client));
             this.commandFactory = Preconditions.CheckNotNull(commandFactory, nameof(commandFactory));
-            this.combinedConfigProvider = Preconditions.CheckNotNull(combinedConfigProvider, nameof(combinedConfigProvider));
+            this.dockerConfigProvider = Preconditions.CheckNotNull(dockerConfigProvider, nameof(dockerConfigProvider));
```

Comment:
Do we still need the dockerconfig?

Reply:
Unfortuanatelly yes. One part of code requires DockerConfig to generate ImagePullSecret based on AuthConfig section. 
Another part that generates EdgeDeploymentDefinition doesn't need the whole AuthCondig section and can use only ImagePullSecretName.

Reply:
How about we pick & choose from dockerconfigprovider and add them in combinedk8sconfig? This way we can manage them better in the long run. Thought?

Reply:
EdgeDeploymentsDefinition has a collection of KubernetesModule objects. CombinedKubernetesConfig. Later EdgeDeploymentsDefinition serialized to a crd command. If we simply put AuthConfig into CombinedKubernetesConfig credentials will be serialized along with module definition. We want to avoid exposing credentials to crd. This is why CombinedKubernetesConfig should not have it.
But we still need to generate ImagePullSecrets based on AuthConfig. That's why I chose to have 2 config providers.
I know it is not ideal, and I open to any valid suggestions.

Reply:
Gave it a second thought and make it more simple.
What before was CombinedKubernetesConfig -> KubernetesConfig
Make CombinedKubernetesConfig be aware of ImagePullSecret